components/Portfolio/AssetDisplay/AssetDisplay.tsx

```diff
@@ -1,36 +1,221 @@
-import type { Asset } from "@/utils/types";
+import type {
+  Asset,
+  AssetCurrent,
+  AssetHistory,
+} from "@/utils/types";
 
-import {
-  useAssetCurrentQueries,
-  useAssetHistoryQueries,
-} from "@/hooks/useAssetQueries";
+import { assetDisplayData } from "@/utils/assetDisplayData";
+import { cn } from "@/utils/cn";
+import { currencyMap } from "@/utils/maps";
 import { extractDate } from "@/utils/extractDate";
+import { localeFormat } from "@/utils/formatHelpers";
+import { useId } from "react";
+
+import CaretIcon from "@/Icons/Caret";
+import Image from "next/image";
+import { Infinity as InfinityIcon } from "lucide-react";
+import ProgressWidget from "@/components/ProgressWidget";
+import { SquarePen as SquarePenIcon } from "lucide-react";
+import { useUserCurrencySetting } from "@/hooks/useUserSettings";
 
 type Props = {
   asset: Asset;
-  currentResponse: ReturnType<typeof useAssetCurrentQueries>[number];
-  historyResponse: ReturnType<typeof useAssetHistoryQueries>[number];
+  assetCurrent: AssetCurrent | undefined;
+  assetHistory: AssetHistory | undefined;
 };
 
-const AssetDisplay = ({ asset, currentResponse, historyResponse }: Props) => {
-  const { coinName, date } = asset;
+const AssetDisplay = ({ asset, assetCurrent, assetHistory }: Props) => {
+  const { coinName, coinSymbol, coinImage, date } = asset;
+  const currency = useUserCurrencySetting();
+
+  const editButtonId = useId();
+  const displayDate = extractDate(date).toLocaleString("en-US", {
+    dateStyle: "medium",
+  });
+
+  const maybeDisplayData = assetDisplayData(
+    asset,
+    currency,
+    assetCurrent,
+    assetHistory
+  );
+
+  const placeholder = <span className="animate-pulse">--</span>;
+
+  const currentValue = (() => {
+    const value = maybeDisplayData?.currentValue;
+    if (!value) return placeholder;
+    return <span>{currencyMap.get(currency) + localeFormat(value)}</span>;
+  })();
+
+  const currentValueChangePercent = (() => {
```

Comment:
"Ternary operator is more concise and can be used in situations where a simple decision needs to be made and assigned to a variable or used within an expression". As the logic is purely determined based on the existing of a value then ternary operator would do the work similar to that in currentValue. I would switch to if...else if logic i complex and have multiple cases to cover not two. However, what you did is also good.